app/scripts/lib/snap-keyring/snap-keyring.ts

```diff
@@ -141,19 +144,29 @@ export const snapKeyringBuilder = (
         const learnMoreLink =
           'https://support.metamask.io/hc/en-us/articles/360015289452-How-to-add-accounts-in-your-wallet';
 
+        // If snap is preinstalled and does not request confirmation, skip the confirmation dialog
+        const skipConfirmation =
+          isSnapPreinstalled(snapId) && !displayConfirmation;
         // Since we use this in the finally, better to give it a default value if the controller call fails
-        let confirmationResult = false;
+        let confirmationResult = skipConfirmation;
+        let addAccountApprovalId = '';
```

Comment:
I feel like `confirmationApprovalId` would be a bit better and "aligned" with the `confirmationResult`, WDYT?

Reply:
Nice call! [44bb44a](https://github.com/MetaMask/metamask-extension/pull/24876/commits/44bb44a2ce8a7f924671ef81cd2efca85d87ee98)